### ISSUE TITLE
Reset test configuration for unit testing after each test (also see #55)

### ISSUE THREAD

**Issue by cyrille-artho** [cyrille-artho/modbat#63]
The configuration management module (`modbat.config.ConfigMgr`) represents all configuration options as variables, which use trait `modbat.config.Configuration`. This makes it very easy to add new options and check the range of parameters at run time. However, for unit testing, this has the problem that overriding the default results in a persistent setting.

We need to be able to reset the configuration options (usually in `modbat.mbt.Configuration`) back to their defaults before each new test. This can be done by adding the ability to clone to trait `modbat.config.Configuration`.

We need

1. a `clone` method for trait `modbat.config.Configuration`;
2. for each type of value, a unit test that clones a test configuration, alters a value on the clone, and ensures that the original still has the correct original setting.

**Comment by cyrille-artho** [comment]
Most configuration items are string, Booleans, ints, or doubles. These are straightforward to clone and test.
The exception are some choice values, which can be strings that refer to integer constants. In this case, you can simply set the right constant in the unit test by looking up where the constants are defined (or ask me).

**Comment by cyrille-artho** [comment]
Here is an outline how to make cloning work, and how to use it:
## Cloning
Modify `modbat.config.Configuration` as follows:
```
trait Configuration extends Cloneable {
  override def clone = super.clone
}
```

## Usage of cloning

Example:
```
class TraitImpl extends Configuration {
  var data = 0
}

object TraitImpl {
  def main(args: Array[String]) {
    val ti = new TraitImpl()
    Console.out.println(ti.data)
    val clone = ti.clone.asInstanceOf[TraitImpl]
    clone.data = 1
    Console.out.println(clone.data)
    Console.out.println(ti.data)
  }
}
```

Write a few unit tests that use assertions rather than print, and test this on a primitive type or two, strings, and choices.

## Use in `ModbatTestHarness` and `Main`:

`Main.run(args)` becomes:
```
val testConfig = Main.config.clone.asInstanceOf[modbat.mbt.Configuration]
Main.run(testConfig, args)
```

`Main.main` and `Main.run` need to support the extra parameter. In `Main.main`: `run(config, args)`.
In `Main.run`, just add another parameter `config: modbat.config.Configuration`. This will then use the parameter instead of the static field. (It may be a good idea to rename the parameter to avoid a name clash.)

**Comment by cyrille-artho** [comment]
For testing `Modbat.mbt.Configuration`, all fields should be straightforward except perhaps `logLevel`. That is an integer, so you can assign any integer constant from `modbat.log.Log` (check in the source code how to import them):
```
val copy = config.clone
copy.logLevel = Log.Fine
assert(config.logLevel == Log.Info) // original value
```
You can write these unit tests in `src/test/modbat/config`.

**Comment by cyrille-artho** [comment]
Mostly resolved in `0444d1084e69aa0f5f66ffc0a61294a5cc8315b5`. Some temporary code (`(args.toList :+ "-s").toArray`) has to be removed at the end, once we get 100 % passing tests on Travis.

**Comment by cyrille-artho** [comment]
Fixed in PR #96
